app/src/main/java/com/cornellappdev/transit/ui/components/RouteCell.kt

```diff
@@ -0,0 +1,682 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Face
+import androidx.compose.material.icons.filled.Place
+import androidx.compose.material.icons.outlined.KeyboardArrowLeft
+import androidx.compose.material.icons.outlined.KeyboardArrowRight
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.cornellappdev.transit.R
+import androidx.constraintlayout.compose.ConstraintLayout
+import com.cornellappdev.transit.ui.theme.IconGrey
+import com.cornellappdev.transit.ui.theme.LateRed
+import com.cornellappdev.transit.ui.theme.LiveGreen
+import com.cornellappdev.transit.ui.theme.MetadataGrey
+import com.cornellappdev.transit.ui.theme.PrimaryText
+import com.cornellappdev.transit.ui.theme.SecondaryText
+import com.cornellappdev.transit.ui.theme.TransitBlue
+import com.cornellappdev.transit.ui.theme.sfProDisplayFamily
+import com.cornellappdev.transit.ui.theme.sfProTextFamily
+
+//Invariant: if transport is WalkOnly, lateness must be None <- maybe can enforce this somehow idk
+@Composable
+fun RouteCell(transport: Transport) {
+    when (transport) {
+        is Transport.WalkOnly -> {
+            Row(
+                modifier = Modifier
+                    .width(375.dp)
+                    .background(color = Color.White, shape = RoundedCornerShape(12.dp))
+            ) {
+                Column(
+                    modifier = Modifier
+                        .padding(16.dp)
+                        .fillMaxWidth()
+                ) {
+                    Row(
+                        horizontalArrangement = Arrangement.SpaceBetween,
+                        modifier = Modifier.fillMaxWidth()
+                    ) {
+                        Text(
+                            "${transport.startTime} - ${transport.arriveTime}",
+                            fontFamily = sfProDisplayFamily,
+                            color = PrimaryText,
+                            fontSize = 16.sp,
+                            fontWeight = FontWeight.SemiBold
+                        )
+
+                        Row(modifier = Modifier.wrapContentWidth()) {
```

Comment:
I see that each of these 3 composables start with this--a surround rounded corner `Row`, a `Column`, and then a timer header at the top. There's a lot of code reuse here as a result.

Instead what we should do is make the `Row` and `Column` in the direct root of the function, and then place the `when` *inside* of the Composable wherever there actually needs to be a different composable.

To reiterate, you're essentially reusing the same `Row` and `Column` and time header code when we can definitely just write them once.